### ISSUE TITLE
opensearch.xml: add documentation about the GET/POST methods.

### ISSUE THREAD

**Issue by dalf** [searx/searx#2069]
https://github.com/asciimoo/searx/blob/6e37255e59693d7a3768491b2b8d06441051131f/searx/templates/__common__/opensearch.xml#L3 
If would be helpful to change this line to ```<ShortName>{{ instance_name }} ( {{opensearch_method}} )</ShortName>```.

It seems a lot of user get confused about the link between the "GET" / "POST" setting and the opensearch.xml behavior.

**Comment by asciimoo** [comment]
I agree, it can be confusing, however I'm not sure that adding the method name here solves the problem. Perhaps better documentation of GET/POST methods would help. Also, there we could emphasize that this whole GET/POST switching nonsense is required because chrome cannot handle POST searches properly.

**Comment by dalf** [comment]
> Perhaps better documentation of GET/POST methods would help.

When an user adds the search engine to the browser, it is not from the preferences page.

The users will either read the documentation because they went the preferences page, or either because the opensearch doesn't work as expected.

So I guess the idea is at least to add an entry about opensearch.xml into the documentation.

**Comment by dalf** [comment]
After a second thought, I think it is more than documentation.

When the browser (at least Firefox) reads opensearch.xml, the cookies are not sent (see https://github.com/asciimoo/searx/pull/2127#issuecomment-671077893 ).
Consequence: the autocomplete setting doesn't work in opensearch.xml (except if it is hard coded in settings.yml).

**Comment by dalf** [comment]
Related to the initial issue: I think an additional change would be helpful, since the cookies are not sent: the /opensearch.xml URL can change according to the settings.
* For example: ```/opensearch.xml?method=GET&autocomplete=duckduckgo```.
* I'm not sure if the query is sent, so an alternative approach can be ```/opensearch.get.duckduckgo.xml```.

The additional benefit: there is no need to erase / force reload to update the opensearch.

**Comment by asciimoo** [comment]
@dalf you are right, probably your suggested method is the good way to handle the issue, especially because the missing cookie thing.

**Comment by dalf** [comment]
With the [multi account container](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/) Firefox extension, the following use case doesn't work:
1. add a searx instance, use the POST method, make searx the default search engine.
1. in a page inside a container, select some text
1. right click "Search <selected text>"

The selected text won't be sent, the index page of searx is shown.

Related issue: https://github.com/mozilla/multi-account-containers/issues/1736

---

I think a clear warning should be displayed the GET method (instead of a link to https://fr.wikipedia.org/wiki/Hypertext_Transfer_Protocol#M.C3.A9thodes )

Example:
1. a user uses Chrome
1. connect to a Google account
1. by default the request are recorded (the title and the URL including the fragment)
1. use searx a the default search engine: request are sent using the GET method since it doesn't work with the POST method. 
1. all requests are stored in the Google account including the queries

There are 2 workarounds:
* don't use Chrome.
* disable recording (on/off button for everything).

Should be the same for other synchronization mechanism like Firefox.

Some technical solutions: 
* ~in addition to ```GET``` and ```POST```, add ```encrypted GET```. The parameters are encrypted with a key made of the client IP and the server timestamp (without the hh:mm:ss). Moreover the HTML title is set using javascript~.
* at least remove the HTML title (see if javascript can set it without being recorded in the browser history)
  * https://github.com/asciimoo/searx/blob/9b9b5e962520b2122cccb32858cce6fc8e0a0130/searx/templates/oscar/results.html#L12
  * https://github.com/asciimoo/searx/blob/9b9b5e962520b2122cccb32858cce6fc8e0a0130/searx/templates/simple/results.html#L3
